### PR TITLE
Ui implement pagination

### PR DESCRIPTION
**Pull request by almoral**

### CODE REVIEW

File: src/app/shared/services/configuration.service.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import {environment} from '../../../environments/environment';
 export class ConfigurationService {
 
   customTitle = environment.customTitle;
+  numberOfLinksToDisplay = 10;
```
Comment (fiallega) [inline src/app/shared/services/configuration.service.ts]: What is number of links to display ? We have to display all pages.

File: src/app/shared/services/data-store.service.ts [hunk 2]
```diff
@@ -37,11 +36,8 @@ export class DataStoreService {
   public textFilter$: Observable<string> = this.textFilterSubject.asObservable();
 
   // observable pagination page number
-  private paginationPageSubject = new BehaviorSubject(1);
-  public paginationPage$: Observable<number> = this.paginationPageSubject.asObservable();
-
-  // observable pagination page number
-  private paginationSubject = new BehaviorSubject<Pagination>(null);
+  private defaultPagination = new Pagination(1, 1, 1);
+  private paginationSubject = new BehaviorSubject<Pagination>(this.defaultPagination);
```
Comment (fiallega) [inline src/app/shared/services/data-store.service.ts]: if this is the only place using defaultPagination, then inline new Pagination(1,1,1). No gain on adding a member variable for that

File: src/app/shared/services/data-store.service.ts [hunk 3]
```diff
@@ -86,24 +82,26 @@ export class DataStoreService {
   getItems() {
     // build filters
     const textFilter = this.textFilterSubject.getValue();
-    const paginationPage = this.paginationPageSubject.getValue();
+    const pagination = this.paginationSubject.getValue();
```
Comment (fiallega) [inline src/app/shared/services/data-store.service.ts]: make the filter variable the currentPage, this.paginationSubject.getValue().currentPage. We dont care about anything else

Comment (almoral) [reply src/app/shared/services/data-store.service.ts]: That works? Lol I didn’t know we could do that. I’ll update it for sure.

File: src/app/index-pagination/index-pagination.component.ts [hunk 4]
```diff
@@ -9,23 +11,41 @@ import { skip } from 'rxjs/operators';
...
     });
 
   }
 
+  setCurrentPage(page: number) {
+    this.dataStoreService.setCurrentPage(page);
+  }
+
+  nextPage() {
+    this.dataStoreService.nextPage();
```
[the remark this answers is not in the document]
Comment (almoral) [reply src/app/index-pagination/index-pagination.component.ts]: Actually the more I think about this it looks like we can't. We need to update the pagination subject and call getItems in the data store.

File: src/app/shared/services/data-store.service.ts (hunk 2 above)
Comment (almoral) [reply src/app/shared/services/data-store.service.ts]: Ok, no problem.

File: package.json [hunk 5]
```diff
@@ -11,12 +11,14 @@
   },
   "private": true,
   "dependencies": {
-    "@angular/animations": "^6.0.2",
+    "@angular/animations": "^6.1.9",
```
[the remark this answers is not in the document]
Comment (almoral) [reply package.json]: This is from using nom audit fix to remove dependencies with security vulnerabilities.

File: src/app/shared/services/configuration.service.ts (hunk 1 above)
Comment (fiallega) [reply src/app/shared/services/configuration.service.ts]: got it.

File: package.json [hunk 6]
```diff
@@ -11,12 +11,14 @@
...
+    "@angular/cdk": "^6.4.7",
     "@angular/common": "^6.0.2",
     "@angular/compiler": "^6.0.2",
     "@angular/core": "^6.0.2",
     "@angular/forms": "^6.0.2",
     "@angular/http": "^6.0.2",
+    "@angular/material": "^6.4.7",
```
[the remark this answers is not in the document]
Comment (fiallega) [reply package.json]: ok

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (fiallega) [reply package.json]: If you move one of them, then move all to 6.1.9

[the remark this answers is not in the document]
Comment (fiallega) [reply package.json]: which I think it is the latest version

**Pull request closed without merging**